### ISSUE TITLE
Spike: explore api.va.gov as possible gateway to expose CMS data

### ISSUE THREAD

**Issue by timcosgrove** [department-of-veterans-affairs/va.gov-cms#14409]
## Description
There are some content data changes that are likely to result in full site rebuilds if they are done through the proposed incremental site regeneration setup. Two identified ones are:

* changes to header or footer menu data
* changes or additions to banners

Ideally we would like the data for these components (and other global components) to be able to update without requiring a full site rebuild, since that would be very costly from a time perspective and would potentially block other content publishing until the full build is complete. 

vets-api is potentially a way for us to allow limited public access to CMS data without doing a full content API. This ticket's scope is to investigate whether this is in fact possible; what would be involved with creating endpoints that draw content from the CMS and expose it to the public; what risks there could be.

This exploration should involve reading vets-api documentation and potentially asking questions of the backend PoC re what we're doing. An actual working PoC is not within scope for this ticket. Other methods of exposing CMS data to the public (i.e. something akin to the content-build KISS mechanism) are also not part of this ticket.

## Acceptance Criteria
- [ ] The process of exposing new API endpoints via api.va.gov is explored and summarized
- [ ] Risks and potential mitigations are identified (dependencies on other teams, any protection that needs to be put on the CMS to protect its data integrity and stability, identify how caching mechanisms work at the API layer, etc)
- [ ] If there are no full blockers, a rough implementation overview is written out in this ticket such that the work can be defined in other tickets

### Team
Please check the team(s) that will do this work.

- [ ] `CMS Team`
- [ ] `Public Websites`
- [ ] `Facilities`
- [ ] `User support`
- [x] `Accelerated publishing`

**Comment by jtmst** [comment]
api.va.gov -> CMS Data Discovery Notes:

### Process for exposing new API endpoints
- New endpoints can be created as standalone endpoints in the main app controller for what seems like smaller one off tasks, or as a standalone module
- Docs point to the preferred method be a generated module.  This would suit our use case better anyways as we will more easily be able to manage additional endpoints should they be required, as well as our drupal config
- Docs for generating EPs within a new module: https://depo-platform-documentation.scrollhelp.site/developer-docs/backend-endpoint-tutorial

### Risks and Mitigations
- External team dependencies:
  - We would be able to develop endpoints ourselves (I have some Rails experience and have additional support from Kind engineers if needed)
  - Platform / COP would need be responsible for maintenance once AP is done.  We'd want to ensure our code is well documented in order to ensure a smooth handoff
- Protection for stability
  - Rate limiting and throttling is easy to configure in vets-api
    - See comment here (config/initializers/rack_attack.rb) for notes on vets-api adding protection following potential facility locator DOS attack
    -  Will need to consider what kind of authentication will be required on the CMS side
 - Caching data for short increments should be relatively simple to handle with redis which is already configured in vets-api

### Other notes
- vets-api does already interact with CMS data in a limited capacity via curl command for the VA forms data: `modules/va_forms/lib/tasks/va_forms.rake`
- We considered using a lambda function for this, but given that most people on our team dont have AWS access, and there doesn't seem to be a caching mechanism already in place, vets-api seems to be the best solution

### Rough implementation outline
- Design API schema with all paths, responses, and errors
- Create module with routing, mocked paths, initial tests
- Implement connection to drupal and logic for any data transformation necessary, caching, response tests
- Documentation (readme and swagger)
- Connect front end with live data

Check out [this](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9626) epic and filter issues by the 'ruby' tag to see a recent example of the stories required to stand up the new Income Limits module in vets-api

**Comment by tjheffner** [comment]
love this, I think you are right that vets-api will be less friction for us (and other teams maintaining) in the long run over AWS. The docs for generating EPs within a new module plus your implementation outline seem like this should be pretty straightforward to set up overall.
